fix: #340 update sphinx and project setup (#341)
canonical/open-documentation-academy: website/docs/howto/get-started/setup_sphinx.md

Type `ls` on the command line to "list show" all the files and directories you have inside your current directory. If this is a fresh Ubuntu virtual machine, you may not have anything yet, except for the `open-documentation-academy` directory that we created when we cloned the repository.

To move into the `open-documentation-academy` directory use `cd` ("change directory").

```bash
cd open-documentation-academy 
```

This will put you into the `open-documentation-academy` directory that contains the current contents of the git repository from GitHub. 

If you're not sure where you are, you can run `ls` again and navigate. Below is an example if you wanted to contribute to the documentation on this website:

```console
user@machine:~/open-documentation-academy$ ls
adsys  charmcraft  charmed-ceph  landscape  LICENSE  multipass  README.md  snapcraft  ubuntu-desktop  website  wsl
user@machine:~/open-documentation-academy$ cd website
```

## Build the documentation

At this point, you can build the documentation (as it currently exists) on your local machine, with the command `make run`.

> `make run` will execute multiple steps, starting with running `install`, and then budiling

Reviewer: ```suggestion
`make run` will execute multiple steps, starting with running `install`, and then building
```

If `make run` doesn't work initially, then try running this sequence of commands to start with a clean environment:

`make clean` : deletes previously locally built documentation files and Python virtual environments. It's safe to run as it does not delete any of the source changes you may have made.
`make install` : creates a Python virtual environment and install all dependencies. This is the recommended way of dealing installing the Python dependencies.

> `make run` : build the documentation and run a local webserver hosting the built html pages.

Reviewer: ```suggestion
`make run` : build the documentation and run a local web server hosting the built HTML pages.
```

If your repository uses Sphinx to create nicely rendered documentation, you will need to set up Sphinx on your machine.
This requires a small chain of steps that are needed to install other things, so you'll need to run through these steps in this order. 

> ## Install System Requirements

Reviewer: ```suggestion
## Install system requirements
```

Sphinx is a Python based documentation framework. To setup an environment, Sphinx needs:

* Python 3
* `pip`, a Python package manager
* `venv`, a Python package that manages [Python virtual enviornments](https://docs.python.org/3/library/venv.html).

> Most projects utilise `make` as a way to automate and abstract operations. Further steps in this guide assume `make` is installed.

Reviewer: ```suggestion
Most projects use `make` as a way to automate and abstract operations. Further steps in this guide assume `make` is installed.
```

On Ubuntu, ensure all dependencies are installed:

```bash
sudo apt update
sudo apt install python3 python3-pip python3-venv make
```

> It's also a good idea to ensure you system is up to date before proceeding with `sudo apt upgrade`

Reviewer: By default I usually suggest `sudo apt update && sudo apt upgrade` to encourage newer users to get into the habit of keeping their system up to date

More advanced users might not want to do that, but this tutorial isn't for them :)

If you've been following along on the previous pages, you should be within a folder (possibly called `src`) that contains a sub-folder called `open-documentation-academy`. Let's check where we are first.

Type `ls` on the command line to "list show" all the files and folders you have inside your current folder. If this is a fresh Ubuntu virtual machine, you may not have anything yet, except for the `open-documentation-academy` folder that we created when we cloned the repository.

> To move into the `open-documentation-academy` directory use `cd` ("change directory").

Reviewer: We're suddenly switching from folder to directory at this point - I think since "directory" is more commonly used we should point out somewhere earlier that directory == folder and then switch to directory throughout

```bash
cd open-documentation-academy 
```

> This will put you into the open-documentation-academy` folder that contains the current contents of the git repository from GitHub. 

Reviewer: ```suggestion
This will put you into the `open-documentation-academy` folder that contains the current contents of the git repository from GitHub. 
```

If you're not sure where you are, you can run `ls` again and navigate. Below is an example if you wanted to contribute to the documentation on this website:

```console
user@machine:~/open-documentation-academy$ ls
adsys  charmcraft  charmed-ceph  landscape  LICENSE  multipass  README.md  snapcraft  ubuntu-desktop  website  wsl
user@machine:~/open-documentation-academy$ cd website
```

## Build the documentation

At this point, you can build the documentation (as it currently exists) on your local machine, with:

```bash

> make run  # builds documentation and runs local webserver hosting the built html pages

Reviewer: The style guide doesn't allow comments inside code blocks - can we move them outside into body text? (splitting the code blocks to allow the comments to be placed appropriately is fine)

```

If `make run` doesn't work initially, then try running this sequence of commands to start with a clean build environment:

```bash

> make clean  # deletes previously built files and Python virtual environment

Reviewer: same here

It's a good idea to open a second Ubuntu tab in your Terminal Window so that you can work in one tab while the documentation can be served in the other. You can do this by clicking on the down arrow next to the currently open tab, and clicking "Ubuntu" (if you're using WSL). 

### On Make commands

Each project may implement different `make` commands. To see available commands run

```bash
make help
```

> `clean` is safe to run as it does not delete any changes you've made to the source code. It only deletes the rendered documentation, making for a "clean" run of `make run`. You can use the `make` commands for more checks depending on the project, such as `make linkcheck` to ensure links are correct and `make spellcheck` for spelling. An example from the `website` makefile

Reviewer: ```suggestion
`clean` is safe to run as it does not delete any changes you've made to the source code. It only deletes the rendered documentation, so that outdated files are not shown on the next build. You can use the `make` commands for more checks depending on the project, such as `make linkcheck` to ensure links are correct and `make spellcheck` for spelling. An example from the `website` makefile
```
